applications/PfemFluidDynamicsApplication/python_scripts/remesh_fluid_domains_process.py

```diff
@@ -165,18 +166,19 @@ def ExecuteInitializeSolutionStep(self):
 
         if(currentStep > 1):
             for domain in self.meshing_domains:
-                domain.ComputeAverageMeshParameters()  
-                meanVolumeBeforeMeshing=domain.GetMeanVolume()
-                totalVolumeBeforeMeshing=domain.GetTotalVolume()
+                if(domain.Active()):
+                    domain.ComputeAverageMeshParameters()  
+                    meanVolumeBeforeMeshing=domain.GetMeanVolume()
+                    totalVolumeBeforeMeshing=domain.GetTotalVolume()
 
-                fileTotalVolume = open("totalVolumeBeforeMeshing.ods", 'a')
-                if(currentStep==2):
-                    fileTotalVolume.seek(0)
-                    fileTotalVolume.truncate()
+                    fileTotalVolume = open("totalVolumeBeforeMeshing.ods", 'a')
```

Comment:
I would avoid opening and closing files at each step/iteration, as it is a relatively expensive operation. You can save the file handle the first time you need to open it and just append to the open file. Also (and this is a minor suggestion) since your file is a standard text file, probably a `.csv` or `.txt` extension would be preferable to an open-office specific extension.